### PR TITLE
Plfm 5795

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java [hunk 1]
```diff
@@ -71,4 +88,128 @@ public static FormGroup createGroupDto(DBOFormGroup dbo) {
...
+		ValidateArgument.required(id, "id");
+		try {
+			return jdbcTemplate.queryForObject("SELECT " + COL_FORM_DATA_GROUP_ID + " FROM " + TABLE_FORM_DATA
+					+ " WHERE " + COL_FORM_DATA_ID + " = ?", String.class, id);
+		} catch (EmptyResultDataAccessException e) {
+			throw new NotFoundException(String.format(FORM_DATA_DOES_NOT_EXIST_FOR_S, id));
+		}
+	}
+
+	@Override
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java:135]: transaction annotation

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java [hunk 2]
```diff
@@ -71,4 +88,128 @@ public static FormGroup createGroupDto(DBOFormGroup dbo) {
...
+		ValidateArgument.required(id, "id");
+		ValidateArgument.required(name, "name");
+		ValidateArgument.required(dataFileHandleId, "dataFileHandleId");
+		jdbcTemplate.update("UPDATE " + TABLE_FORM_DATA + " SET " + COL_FORM_DATA_NAME + " = ?, "
+				+ COL_FORM_DATA_FILE_ID + " = ?, " + COL_FORM_DATA_ETAG + " = UUID(), " + COL_FORM_DATA_MODIFIED_ON
+				+ " = NOW(3) WHERE " + COL_FORM_DATA_ID + " = ?", name, dataFileHandleId, id);
+		return getFormData(id);
+	}
+
+	@Override
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java:146]: transaction

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImplTest.java [hunk 3]
```diff
@@ -84,4 +118,315 @@ public void testLookupGroupByName() {
...
+		assertEquals("FormData does not exist for: -1", message);
+	}
+
+	@Test
+	public void testGetFormDataState() {
+		FormData data = createFormData();
+		// call under test
+		StateEnum state = formDao.getFormDataState(data.getFormDataId());
+		assertEquals(data.getSubmissionStatus().getState(), state);
+	}
```
Comment (john-hill) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImplTest.java:262]: missing state not found test

**Review by marcomarasca — APPROVED** [review]
LGTM

**Pull request merged**